src/java/org/apache/cassandra/locator/TokenMetadata.java

```diff
@@ -904,38 +905,38 @@ private static PendingRangeMaps calculatePendingRanges(AbstractReplicationStrate
         for (Pair<Token, InetAddressAndPort> moving : movingEndpoints)
         {
             //Calculate all the ranges which will could be affected. This will include the ranges before and after the move.
-            Set<Range<Token>> moveAffectedRanges = new HashSet<>();
+            Set<Replica> moveAffectedReplicas = new HashSet<>();
```

Comment:
Need to think carefully about this usage of Set<Replica> move is the case where endpoints and transientness changes but ranges might not. That's when it breaks down to try and use sets and set difference of all 3 attributes.